libraries/botbuilder-core/src/storage.ts

```diff
@@ -115,10 +116,29 @@ export const assertStoreItems: Assertion<StoreItems> = (val, path) => {
  * @param item Item to calculate the change hash for.
  */
 export function calculateChangeHash(item: StoreItem): string {
-    const cpy: any = { ...item };
-    if (cpy.eTag) {
-        delete cpy.eTag;
+    if (!item) {
+        return '';
     }
 
-    return JSON.stringify(cpy);
+    const { eTag, ...rest } = item;
```

Comment:
Would be a good idea to catch circular references errors and only in those cases apply all this logic?

Reply:
Yea, it would be a good idea to have something like that to detect circular references before calling this function, but there is no builtin or something related.
We can create a function that does that (image), but at the end of the day it will iterate the object.
Maybe we can create a minimal function that check if the object has circular references and end the iteration in the first circular reference it finds.
![image](https://github.com/southworks/botbuilder-js/assets/62260472/367bbdce-861a-4650-8221-b27c0ff76c92)

Reply:
I was thinking on a simple try-catch for the specific exception.

Reply:
Ok, i will do that

Reply:
Updated thanks!